terraform/README.md

```diff
@@ -41,30 +41,43 @@ This Terraform setup provisions an AWS infrastructure to support an ETL pipeline
 - **AWS**: Configured to deploy resources in the `eu-west-2` region.
 
 ### Security Groups
-- A security group (`ecs_task_sg`) is created to allow:
+- **ETL Security Group (`ecs_task_sg`)**: Allows:
   - Inbound traffic on port `1433` (SQL Server).
   - Inbound traffic on port `433` (API).
   - All outbound traffic.
+- **Dashboard Security Group (`ecs_service_sg`)**: Allows:
+  - Inbound traffic on port `8501` (Streamlit app).
+  - All outbound traffic.
 
-### ECR Repository
-- An ECR repository (`c14-team-growth-lmnh`) is provisioned for storing Docker images for the ETL pipeline.
+### ECR Repositories
+- **ETL Repository (`c14-team-growth-lmnh`)**: Stores Docker images for the ETL pipeline.
+- **Dashboard Repository (`c14-team-growth-lmnh-dashboard`)**: Stores Docker images for the Streamlit dashboard.
 
 ### IAM Roles and Policies
 - **ECS Task Role**: Grants permissions to upload files to S3.
+- **ECS Service Role**: Grants permissions to access S3 resources required by the dashboard.
 - **Lambda Role**: Includes policies for basic execution, VPC access, and RDS access.
 - **EventBridge Role**: Allows EventBridge to trigger ECS tasks and Lambda functions.
 
-### ECS Task Definition
-- An ECS Fargate task (`c14-team-growth-rds-to-s3-etl`) runs the ETL container with the required CPU, memory, and environment variables (e.g., `DB_HOST`, `DB_PORT`, `S3_BUCKET`).
+### ECS Task Definitions
+- **ETL Task (`c14-team-growth-rds-to-s3-etl`)**: Runs the ETL container with environment variables like `DB_HOST`, `S3_BUCKET`, etc.
+- **Dashboard Task (`c14-team-growth-dashboard`)**: Runs the Streamlit dashboard with port mappings for `8501`.
+
+### ECS Service
```

Comment:
Consider specifying whether the Streamlit dashboard is deployed in a public or private subnet, and if public, how it's secured (e.g., using a load balancer). This will help users understand network accessibility.

Reply:
This is a good point, I think unnecessary for the moment as there is no load balancer used and users can easily access the board.